tests/vxlan/test_vxlan_ecmp.py

```diff
@@ -630,3 +637,408 @@ def test_vxlan_single_endpoint(self, setUp, encap_type):
+        for i in range(3):
+            tc11_new_nhs.append(get_ip_address(af=get_outer_layer_version(encap_type), netid=NextHop_Prefix))
+
+        # the topology always provides the default routes for any ip address.
+        # so it is already taken care of.
+
+        # Map the new dest and nhs.
+        tc11_config = '[\n' + create_single_route(vnet, tc11_new_dest, HOST_MASK[get_payload_version(encap_type)], tc11_new_nhs, "SET") + '\n]'
+        self.setup[encap_type]['dest_to_nh_map'][vnet][tc11_new_dest] = tc11_new_nhs
+
+        # Apply the config in the DUT and verify traffic. The random hash and ECMP check is already taken care of in the VxLAN PTF script.
+        apply_config_in_swss(self.setup['duthost'], tc11_config, "vnet_route_tc11_"+encap_type)
+        self.dump_self_info_and_run_ptf("tc11", encap_type, True, packet_count=1000)
+
+class Test_VxLAN_underlay_ecmp(Test_VxLAN):
+    @pytest.mark.parametrize("ecmp_path_count", [1, 2])
+    def test_vxlan_modify_underlay_default(self, setUp, minigraph_facts, encap_type, ecmp_path_count):
+        self.setup = setUp
+        Logger.info("tc12: modify the underlay default route nexthop/s. send packets to route 3's prefix dst.")
+        # First step: pick one or two of the interfaces connected to t2, and bring them down.
+        # verify that the encap is still working, and ptf receives the traffic.
+        # Bring them back up.
+        # After that, bring down all the other t2 interfaces, other than the ones used in the first step.
+        # This will force a modification to the underlay default routes nexthops.
+
+        all_intfs = list(get_portchannels_to_neighbors(self.setup['duthost'], "T2", minigraph_facts))
+        if not all_intfs:
+            all_intfs = get_ethernet_to_neighbors("T2", minigraph_facts)
+        Logger.info("Dumping T2 link info: {}".format(all_intfs))
+        if not all_intfs:
+            raise RuntimeError("no interface found connected to t2 neighbors. pls check the testbed, aborting.")
+
+        # Backup
+        all_t2_ports = list(self.setup[encap_type]['t2_ports'])
+        try:
+            shut_intfs = []
+            # Choose some intfs based on the parameter ecmp_path_count.
+            # when ecmp_path_count == 1, it is non-ecmp. The switching happens between ecmp and non-ecmp.
+            # Otherwise, the switching happens within ecmp only.
+            for i in range(ecmp_path_count):
+                shut_intfs.append(all_intfs[i])
+
+            for intf in shut_intfs:
+                self.setup['duthost'].shell("sudo config interface shutdown {}".format(intf))
+            self.dump_self_info_and_run_ptf("tc12", encap_type, True, packet_count=1000)
+
+            # Bring up the selected intfs.
+
+            self.setup[encap_type]['t2_ports'] = list(all_t2_ports)
+            for intf in shut_intfs:
+                self.setup['duthost'].shell("sudo config interface startup {}".format(intf))
+            time.sleep(60)
```

Comment:
sleep is flaky and not efficient, please replace it with 'wait until'

Reply:
Done.